schemas/json/theme.json

```diff
@@ -226,6 +226,69 @@
 								"type": "string"
 							},
 							"default": [ "px", "em", "rem", "vh", "vw", "%" ]
+						},
+						"customSpacingSize": {
+							"description": "Allow users to set custom space sizes.",
+							"type": "boolean",
+							"default": true
+						},
+						"spacingSizes": {
+							"description": "Space size presets for the space size selector.\nGenerates a custom property (`--wp--preset--space-size--{slug}`) per preset value.",
+							"type": "array",
+							"items": {
+								"type": "object",
+								"properties": {
+									"name": {
+										"description": "Name of the space size preset, translatable.",
+										"type": "string"
+									},
+									"slug": {
+										"description": "Kebab-case unique identifier for the space size preset.",
+										"type": "string"
+									},
+									"size": {
+										"description": "CSS space-size value, including units.",
+										"type": "string"
+									}
+								},
+								"additionalProperties": false
+							}
+						},
+						"spacingScale": {
+							"description": "Space size presets for the space size selector.\nGenerates a custom property (`--wp--preset--space-size--{slug}`) per preset value.",
+							"type": "object",
+							"properties": {
+								"operator": {
+									"description": "With + or * depending on whether scale is generated by increment or mulitplier.",
+									"type": "string",
+									"enum": [ "+", "*" ]
+								},
+								"increment": {
+									"description": "The amount to increment each step by.",
+									"type": "number"
+								},
+								"steps": {
+									"description": "Number of steps to generate in scale.",
+									"type": "integer"
```

Comment:
In my testing so far, the steps generate equals _one less_ than the value of `step`.

So for `"step": 2` I see:
```css
	--wp--preset--spacing--50: 1.5rem;
```

`"step": 4`:

```css
	--wp--preset--spacing--10: 0rem;
	--wp--preset--spacing--50: 1.5rem;
	--wp--preset--spacing--30: 3rem;
```


And so on.

Am I doing something wrong or is this description not right?

Reply:
looks like a classic `for loop` out by 1 error doesn't it - will take a look.

Reply:
Thanks @ramonjd , nice catch, it was indeed a case of `ye olde out by one error`. It also highlighted another issue in that having a mid point of 50 means we couldn't have a `steps:11` without pushing the  bottom slugs into the minuses. I have adjusted so `steps` > 10 stops at bottom slug of 10 and adds additional items to top of scale.  Also for even numbers it currently adds the odd item above the midpoint of 50 eg. `steps: 4` has `40,50,60,70` rather than `30,40,50,60`  - I figure people are more likely to want more larger space options than smaller - but let me know if you think the reverse makes more sense.

Reply:
Would it be worth adding some unit tests so we have a record of what to expect?

I'm fiddling around with spacing presets and the CSS vars are auto-generated and printed to the page, but I'm not sure if what I'm seeing is expected or not.

Example:

```json
"spacingScale": {
    "increment": 0.25,
    "steps": 4,
    "mediumStep": 2,
    "units": "rem"
}
```

Gives me:

```css
--wp--preset--spacing--10: 512rem;
--wp--preset--spacing--20: 128rem;
--wp--preset--spacing--40: 8rem;
--wp--preset--spacing--50: 2rem;
--wp--preset--spacing--60: 0.5rem;
--wp--preset--spacing--70: 0.13rem;
```


And

```json
"spacingScale": {
	"operator": "+",
	"increment": 1.5,
	"steps": 10,
	"mediumStep": 4,
	"units": "rem"
}
```

Prints out:

```css
--wp--preset--spacing--10: -2rem;
--wp--preset--spacing--20: -0.5rem;
--wp--preset--spacing--30: 1rem;
--wp--preset--spacing--40: 2.5rem;
--wp--preset--spacing--50: 4rem;
--wp--preset--spacing--60: 5.5rem;
--wp--preset--spacing--70: 7rem;
--wp--preset--spacing--80: 8.5rem;
--wp--preset--spacing--90: 10rem;
--wp--preset--spacing--100: 11.5rem;
```

It could be all fine 😄 I don't know.

If the number of steps generated doesn't correspond to the value of `steps` should we update the schema description to `Minimum number of steps to generate in scale.` or something?

Reply:
Thanks @ramon, there are a couple of things I forgot to account for - will sort in the morning and add some tests